### PR TITLE
onetbb: add 2021.12.0

### PR DESCRIPTION
**Pull request by Nekto89**
Specify library name and version:  **onetbb/2021.12.0**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by valgur — APPROVED** [review]

File: recipes/onetbb/all/conanfile.py [hunk 1]
```diff
@@ -120,7 +120,7 @@ def generate(self):
         toolchain.variables["TBB_DISABLE_HWLOC_AUTOMATIC_SEARCH"] = not self._tbbbind_build
         if self._tbbbind_explicit_hwloc:
             hwloc_package_folder = self.dependencies["hwloc"].package_folder
-            hwloc_lib_name = "hwloc.lib" if self.settings.os == "Windows" else "libhwloc.so"
+            hwloc_lib_name = "hwloc.lib" if self.settings.os == "Windows" else "libhwloc.dylib" if self.settings.os == "Macos" else "libhwloc.so"
```
Comment (valgur) [inline recipes/onetbb/all/conanfile.py:123]: I would split it into a multi-line statement.

Comment (Nekto89) [reply recipes/onetbb/all/conanfile.py]: done

**Review by RubenRBS — APPROVED** [review]
Thanks a lot!

**Review by toge — APPROVED** [review]

**Comment by smanolloff** [comment]
Hello everyone,

is it possible that something went wrong with the rollout of this recipe that could cause **previous recipe versions to be overwritten**?

The issue I am observing is: all onetbb versions 2021.x contain a hwloc dependency.
This surely was not the case several months ago. I had some cached versions of 2021.08 which did not have a hwloc dependency, but re-downloading the recipe introduced it.

Additionally, [the information in conan.io is wrong](https://conan.io/center/recipes/onetbb) -- it says only 2021.12 has a `hwloc` dependency, but that is not true. All 2021.x have this dependency now.

```bash
# obtain download url for 2021.8.0
curl 'https://center.conan.io/v1/conans/onetbb/2021.8.0/_/_/download_urls'

# download recipe for 2021.8.0
curl 'https://center.conan.io/artifactory/api/conan/conan-center/v1/files/_/onetbb/2021.8.0/_/fa6894d676f63db5f9dad05298a8eb0e/export/conanfile.py'

# inspect the recipe and confirm hwloc dependency
# (same for all 2021.x versions)
```

I noticed it since `hwloc` is a no-go for us (does not compile for iOS) and we have pinned `onetbb` to a specific version that is not supposed to depend on it, however our builds have started failing now (we may have been running on a cached recipe for months).

**Comment by smanolloff** [comment]
All sorted out in https://github.com/conan-io/conan-center-index/issues/25337. Ignore my previous comment here.

**Pull request merged**